### PR TITLE
[EOC-256] SOCKET Deleting members from the list

### PR DESCRIPTION
**Pull request by adamklepacz**
This branch includes changes from Aleksander's `eoc_264` branch and from other branches cause `eoc_264` includes some other branches. Don't know why? (@alemikolo ??)

Anyway, my changes to review, you can find in those files: 
```
client/src/modules/list/model/actions.js 
client/src/sockets/helpers/index.js
server/sockets/index.js
server/sockets/list.js
```

### CODE REVIEW

File: client/src/sockets/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+      if (cohortId) {
+        dispatch({ type: ListEvents.DELETE_SUCCESS, payload: listId });
+
+        return history.replace(`/cohort/${cohortId}`);
+      }
+
+      dispatch({ type: ListEvents.DELETE_SUCCESS, payload: listId });
+
+      return history.replace('/dashboard');
```
Comment (alemikolo) [inline client/src/sockets/helpers/index.js]: You can simplify this: 
```
case ListEvents.REMOVE_BY_SOMEONE: {
      const { cohortId, listId } = data;

      dispatch({ type: ListEvents.DELETE_SUCCESS, payload: listId });

      return history.replace(
        `/${cohortId ? `cohort/${cohortId}` : 'dashboard'}`
      );
    }
```

File: server/sockets/helpers/index.js [hunk 2]
```diff
@@ -43,37 +46,35 @@ const updateListOnDashboardAndCohortView = (
...
-              // send to users that are on the dashboard view
-              socket.broadcast
-                .to(dashboardClients.get(viewerId))
-                .emit(ListActionTypes.CREATE_SUCCESS, list);
-            }
-          });
-        }
+          if (dashboardClients.has(viewerId)) {
+            const { socketId } = dashboardClients.get(viewerId);
+            // send to users that are on the dashboard view
```
Comment (alemikolo) [inline server/sockets/helpers/index.js]: this comment is unnecessary, we use dashboardClients here so we know that this event will be sendt to users that are on the dashboard

File: server/sockets/index.js [hunk 3]
```diff
@@ -17,12 +17,14 @@ const {
...
   archiveItem,
   changeItemOrderState,
+  changeListType,
   clearVote,
   cloneItem,
   deleteItem,
   emitListsOnAddCohortMember,
   emitRemoveMemberOnLeaveCohort,
   leaveList,
+  removeMember,
```
Comment (alemikolo) [inline server/sockets/index.js]: removeListMember ?

File: server/sockets/index.js [hunk 4]
```diff
@@ -126,13 +129,25 @@ const socketListenTo = server => {
...
+      cohortViewClients,
+      listViewClients
+    );
     clearVote(socket);
     cloneItem(socket);
     deleteItem(socket);
     emitListsOnAddCohortMember(socket, dashboardViewClients);
     emitRemoveMemberOnLeaveCohort(socket);
     leaveList(socket);
+    removeMember(
```
Comment (alemikolo) [inline server/sockets/index.js]: removeListMember ?

File: server/sockets/list.js [hunk 5]
```diff
@@ -348,6 +384,151 @@ const emitRemoveMemberOnLeaveCohort = socket =>
...
+          });
+
+          socket.broadcast
+            .to(`sack-${listId}`)
+            .emit(ListActionTypes.CHANGE_TYPE_SUCCESS, data);
+        }
+      });
+  });
+
+const removeMember = (socket, dashboardClients, listClients, cohortClients) =>
```
Comment (alemikolo) [inline server/sockets/list.js]: removeListMember

**Comment by alemikolo** [comment]
You have to change the base from master to eoc_264...

**Comment by adamklepacz** [comment]
> You have to change the base from master to eoc_264...

It doesn't matter if you have all files pointed in the first comment.

I will merge it to the master anyway

File: server/sockets/list.js [hunk 6]
```diff
@@ -348,6 +384,156 @@ const emitRemoveMemberOnLeaveCohort = socket =>
...
+const removeListMember = (
+  socket,
+  dashboardClients,
+  listClients,
+  cohortClients
+) =>
+  socket.on(ListActionTypes.REMOVE_MEMBER_SUCCESS, data => {
+    const { listId, userId } = data;
+
+    // Broadcast to that removed user if he is at dashboard view
```
Comment (alemikolo) [inline server/sockets/list.js]: remove comments, dashboardClients in use so it's clear that you emit to users on dashbaord

File: server/sockets/list.js [hunk 7]
```diff
@@ -348,6 +384,156 @@ const emitRemoveMemberOnLeaveCohort = socket =>
...
+            const { socketId, viewId } = listClients.get(userId);
+            const { cohortId } = doc;
+            const data = { listId };
+
+            if (viewId === listId) {
+              if (cohortId) {
+                data.cohortId = cohortId;
+              }
+
+              // Broadcast to that removed user if he is on the list view
```
Comment (alemikolo) [inline server/sockets/list.js]: remove comment you use listClients and compare viewId === listId so it's clear that user is on the list view

File: server/sockets/list.js [hunk 8]
```diff
@@ -348,6 +384,156 @@ const emitRemoveMemberOnLeaveCohort = socket =>
...
+              socket.broadcast
+                .to(socketId)
+                .emit(ListActionTypes.REMOVE_BY_SOMEONE, data);
+            }
+          }
+        });
+    }
+
+    // Broadcast to that removed user if he is on the cohort view
```
Comment (alemikolo) [inline server/sockets/list.js]: Our Maps' names (Maps which store clients id, viewId, and socketId) are unequivocal. If we use their names we don't have to use comments

File: server/sockets/list.js [hunk 9]
```diff
@@ -348,6 +384,156 @@ const emitRemoveMemberOnLeaveCohort = socket =>
...
+      const { socketId } = cohortClients.get(userId);
+
+      socket.broadcast
+        .to(socketId)
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+
+    // Broadcast to every user on the list
+    socket.broadcast
+      .to(`sack-${listId}`)
```
Comment (alemikolo) [inline server/sockets/list.js]: this line says: 'send to every user on the list' so I think the comment is unnecessary

Comment (alemikolo) [inline server/sockets/list.js]: this line says: 'send to every user on the list' so I think the comment is unnecessary

**Pull request merged**